[brace-style] doesn't check enum

Given the rule
```JSON
{
  "rules": {
    "@typescript-eslint/brace-style": ["error", "allman"]
  }
}
```

I am expecting to receive a warning on the open brace for
```typescript
export enum AnEnum {
    SOMETHING = "something",
    ELSE = "esle",
}
```

but I don't

**Versions**
| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.9.0` |
| `@typescript-eslint/parser`        | `2.9.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |
| `yarn`                              | `1.19.1` |

**Debug Dump**
```
~/application $ eslint -c ./.eslintrc.js --ext ".ts" ./packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts --debug
  eslint:cli CLI args: [ '-c', './.eslintrc.js', '--ext', '.ts', './packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts', '--debug' ] +0ms
  eslint:cli Running on files +6ms
  eslint:config-array-factory Loading JS config file: /home/nodebrick/application/.eslintrc.js +0ms
  eslint:config-array-factory Loading {extends:"eslint:recommended"} relative to /home/nodebrick/application/.eslintrc.js +22ms
  eslint:config-array-factory Loading JS config file: /home/nodebrick/application/node_modules/eslint/conf/eslint-recommended.js +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/eslint-recommended"} relative to /home/nodebrick/application/.eslintrc.js +13ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /home/nodebrick/application/.eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.9.0 (/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +11ms
  eslint:config-array-factory Plugin /home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 785ms +785ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /home/nodebrick/application/.eslintrc.js +2ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.9.0 (/home/nodebrick/application/node_modules/@typescript-eslint/parser/dist/parser.js) +9ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /home/nodebrick/application/.eslintrc.js +239ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.9.0 (/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Plugin /home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:ignored-paths baseDir = "/home/nodebrick/application" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/node_modules/*"
  eslint:ignored-paths   cooked   = "/node_modules/*" +1ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/bower_components/*"
  eslint:ignored-paths   cooked   = "/bower_components/*" +1ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = ".*"
  eslint:ignored-paths   cooked   = ".*" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "!../"
  eslint:ignored-paths   cooked   = "!../" +0ms
  eslint:ignored-paths Looking for ignore file in /home/nodebrick/application +0ms
  eslint:ignored-paths Could not find ignore file in cwd +1ms
  eslint:ignored-paths baseDir = "/home/nodebrick/application" +4ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/node_modules/*"
  eslint:ignored-paths   cooked   = "/node_modules/*" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/bower_components/*"
  eslint:ignored-paths   cooked   = "/bower_components/*" +0ms
  eslint:ignored-paths Looking for ignore file in /home/nodebrick/application +0ms
  eslint:ignored-paths Could not find ignore file in cwd +0ms
  eslint:file-enumerator Start to iterate files: [ './packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts' ] +0ms
  eslint:file-enumerator File: /home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts +3ms
  eslint:cascading-config-array-factory Load config files for /home/nodebrick/application/packages/nodebrick-database/src/models/enums. +0ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick/application/packages/nodebrick-database/src/models/enums. +0ms
  eslint:config-array-factory Config file not found on /home/nodebrick/application/packages/nodebrick-database/src/models/enums +17ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick/application/packages/nodebrick-database/src/models. +2ms
  eslint:config-array-factory Config file not found on /home/nodebrick/application/packages/nodebrick-database/src/models +2ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick/application/packages/nodebrick-database/src. +2ms
  eslint:config-array-factory Config file not found on /home/nodebrick/application/packages/nodebrick-database/src +2ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick/application/packages/nodebrick-database. +2ms
  eslint:config-array-factory Loading package.json config file: /home/nodebrick/application/packages/nodebrick-database/package.json +3ms
  eslint:config-array-factory Loading JSON config file: /home/nodebrick/application/packages/nodebrick-database/package.json +0ms
  eslint:config-array-factory Error reading package.json file: /home/nodebrick/application/packages/nodebrick-database/package.json +3ms
  eslint:config-array-factory Config file not found on /home/nodebrick/application/packages/nodebrick-database +0ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick/application/packages. +6ms
  eslint:config-array-factory Config file not found on /home/nodebrick/application/packages +2ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick/application. +2ms
  eslint:config-array-factory Loading JS config file: /home/nodebrick/application/.eslintrc.js +1ms
  eslint:config-array-factory Config file found: /home/nodebrick/application/.eslintrc.js +5ms
  eslint:config-array-factory Loading {extends:"eslint:recommended"} relative to /home/nodebrick/application/.eslintrc.js +1ms
  eslint:config-array-factory Loading JS config file: /home/nodebrick/application/node_modules/eslint/conf/eslint-recommended.js +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/eslint-recommended"} relative to /home/nodebrick/application/.eslintrc.js +5ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /home/nodebrick/application/.eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.9.0 (/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +0ms
  eslint:config-array-factory Plugin /home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /home/nodebrick/application/.eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.9.0 (/home/nodebrick/application/node_modules/@typescript-eslint/parser/dist/parser.js) +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /home/nodebrick/application/.eslintrc.js +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.9.0 (/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +0ms
  eslint:config-array-factory Plugin /home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 0ms +0ms
  eslint:cascading-config-array-factory No cache found: /home/nodebrick. +13ms
  eslint:cascading-config-array-factory Stop traversing because of considered root. +0ms
  eslint:rules Loading rule 'constructor-super' (remaining=269) +0ms
  eslint:rules Loading rule 'for-direction' (remaining=268) +6ms
  eslint:rules Loading rule 'getter-return' (remaining=267) +3ms
  eslint:rules Loading rule 'no-async-promise-executor' (remaining=266) +7ms
  eslint:rules Loading rule 'no-case-declarations' (remaining=265) +3ms
  eslint:rules Loading rule 'no-class-assign' (remaining=264) +4ms
  eslint:rules Loading rule 'no-compare-neg-zero' (remaining=263) +4ms
  eslint:rules Loading rule 'no-cond-assign' (remaining=262) +3ms
  eslint:rules Loading rule 'no-const-assign' (remaining=261) +4ms
  eslint:rules Loading rule 'no-constant-condition' (remaining=260) +5ms
  eslint:rules Loading rule 'no-control-regex' (remaining=259) +5ms
  eslint:rules Loading rule 'no-debugger' (remaining=258) +22ms
  eslint:rules Loading rule 'no-delete-var' (remaining=257) +4ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=256) +8ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=255) +9ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=254) +4ms
  eslint:rules Loading rule 'no-duplicate-case' (remaining=253) +6ms
  eslint:rules Loading rule 'no-empty' (remaining=252) +5ms
  eslint:rules Loading rule 'no-empty-character-class' (remaining=251) +6ms
  eslint:rules Loading rule 'no-empty-pattern' (remaining=250) +4ms
  eslint:rules Loading rule 'no-ex-assign' (remaining=249) +6ms
  eslint:rules Loading rule 'no-extra-boolean-cast' (remaining=248) +5ms
  eslint:rules Loading rule 'no-extra-semi' (remaining=247) +4ms
  eslint:rules Loading rule 'no-fallthrough' (remaining=246) +5ms
  eslint:rules Loading rule 'no-func-assign' (remaining=245) +5ms
  eslint:rules Loading rule 'no-global-assign' (remaining=244) +6ms
  eslint:rules Loading rule 'no-inner-declarations' (remaining=243) +6ms
  eslint:rules Loading rule 'no-invalid-regexp' (remaining=242) +5ms
  eslint:rules Loading rule 'no-irregular-whitespace' (remaining=241) +5ms
  eslint:rules Loading rule 'no-misleading-character-class' (remaining=240) +5ms
  eslint:rules Loading rule 'no-mixed-spaces-and-tabs' (remaining=239) +29ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=238) +3ms
  eslint:rules Loading rule 'no-obj-calls' (remaining=237) +4ms
  eslint:rules Loading rule 'no-octal' (remaining=236) +4ms
  eslint:rules Loading rule 'no-prototype-builtins' (remaining=235) +3ms
  eslint:rules Loading rule 'no-redeclare' (remaining=234) +3ms
  eslint:rules Loading rule 'no-regex-spaces' (remaining=233) +5ms
  eslint:rules Loading rule 'no-self-assign' (remaining=232) +3ms
  eslint:rules Loading rule 'no-shadow-restricted-names' (remaining=231) +6ms
  eslint:rules Loading rule 'no-sparse-arrays' (remaining=230) +4ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=229) +4ms
  eslint:rules Loading rule 'no-undef' (remaining=228) +4ms
  eslint:rules Loading rule 'no-unexpected-multiline' (remaining=227) +5ms
  eslint:rules Loading rule 'no-unreachable' (remaining=226) +5ms
  eslint:rules Loading rule 'no-unsafe-finally' (remaining=225) +5ms
  eslint:rules Loading rule 'no-unsafe-negation' (remaining=224) +3ms
  eslint:rules Loading rule 'no-unused-labels' (remaining=223) +4ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=222) +5ms
  eslint:rules Loading rule 'no-useless-catch' (remaining=221) +1ms
  eslint:rules Loading rule 'no-useless-escape' (remaining=220) +4ms
  eslint:rules Loading rule 'no-with' (remaining=219) +4ms
  eslint:rules Loading rule 'require-atomic-updates' (remaining=218) +4ms
  eslint:rules Loading rule 'require-yield' (remaining=217) +3ms
  eslint:rules Loading rule 'use-isnan' (remaining=216) +4ms
  eslint:rules Loading rule 'valid-typeof' (remaining=215) +6ms
  eslint:rules Loading rule 'brace-style' (remaining=214) +4ms
  eslint:rules Loading rule 'camelcase' (remaining=213) +0ms
  eslint:rules Loading rule 'indent' (remaining=212) +1ms
  eslint:rules Loading rule 'no-empty-function' (remaining=211) +0ms
  eslint:rules Loading rule 'no-extra-parens' (remaining=210) +0ms
  eslint:rules Loading rule 'no-magic-numbers' (remaining=209) +0ms
  eslint:rules Loading rule 'no-unused-expressions' (remaining=208) +1ms
  eslint:rules Loading rule 'quotes' (remaining=207) +0ms
  eslint:rules Loading rule 'require-await' (remaining=206) +0ms
  eslint:rules Loading rule 'semi' (remaining=205) +0ms
  eslint:rules Loading rule 'space-before-function-paren' (remaining=204) +1ms
  eslint:rules Loading rule 'arrow-body-style' (remaining=203) +5ms
  eslint:rules Loading rule 'arrow-parens' (remaining=202) +6ms
  eslint:rules Loading rule 'capitalized-comments' (remaining=201) +6ms
  eslint:rules Loading rule 'no-multi-str' (remaining=200) +9ms
  eslint:rules Loading rule 'no-multiple-empty-lines' (remaining=199) +4ms
  eslint:rules Loading rule 'no-param-reassign' (remaining=198) +4ms
  eslint:rules Loading rule 'space-in-parens' (remaining=197) +6ms
  eslint:cascading-config-array-factory Configuration was determined: ConfigArray [ { name: '.eslintrc.js » eslint:recommended', filePath: '/home/nodebrick/application/node_modules/eslint/conf/eslint-recommended.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'constructor-super': 'error', 'for-direction': 'error', 'getter-return': 'error', 'no-async-promise-executor': 'error', 'no-case-declarations': 'error', 'no-class-assign': 'error', 'no-compare-neg-zero': 'error', 'no-cond-assign': 'error', 'no-const-assign': 'error', 'no-constant-condition': 'error', 'no-control-regex': 'error', 'no-debugger': 'error', 'no-delete-var': 'error', 'no-dupe-args': 'error', 'no-dupe-class-members': 'error', 'no-dupe-keys': 'error', 'no-duplicate-case': 'error', 'no-empty': 'error', 'no-empty-character-class': 'error', 'no-empty-pattern': 'error', 'no-ex-assign': 'error', 'no-extra-boolean-cast': 'error', 'no-extra-semi': 'error', 'no-fallthrough': 'error', 'no-func-assign': 'error', 'no-global-assign': 'error', 'no-inner-declarations': 'error', 'no-invalid-regexp': 'error', 'no-irregular-whitespace': 'error', 'no-misleading-character-class': 'error', 'no-mixed-spaces-and-tabs': 'error', 'no-new-symbol': 'error', 'no-obj-calls': 'error', 'no-octal': 'error', 'no-prototype-builtins': 'error', 'no-redeclare': 'error', 'no-regex-spaces': 'error', 'no-self-assign': 'error', 'no-shadow-restricted-names': 'error', 'no-sparse-arrays': 'error', 'no-this-before-super': 'error', 'no-undef': 'error', 'no-unexpected-multiline': 'error', 'no-unreachable': 'error', 'no-unsafe-finally': 'error', 'no-unsafe-negation': 'error', 'no-unused-labels': 'error', 'no-unused-vars': 'error', 'no-useless-catch': 'error', 'no-useless-escape': 'error', 'no-with': 'error', 'require-atomic-updates': 'error', 'require-yield': 'error', 'use-isnan': 'error', 'valid-typeof': 'error' }, settings: undefined }, { name: '.eslintrc.js » plugin:@typescript-eslint/eslint-recommended', filePath: '/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:@typescript-eslint/eslint-recommended#overrides[0]', filePath: '/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: { includes: [Array], excludes: null, basePath: '/home/nodebrick/application' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'getter-return': 'off', 'no-dupe-args': 'off', 'no-dupe-keys': 'off', 'no-unreachable': 'off', 'valid-typeof': 'off', 'no-const-assign': 'off', 'no-new-symbol': 'off', 'no-this-before-super': 'off', 'no-undef': 'off', 'no-dupe-class-members': 'off', 'no-redeclare': 'off' }, settings: undefined }, { name: '.eslintrc.js', filePath: '/home/nodebrick/application/.eslintrc.js', criteria: null, env: { node: true, es6: true }, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2018 }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: {}, settings: undefined }, { name: '.eslintrc.js#overrides[0]', filePath: '/home/nodebrick/application/.eslintrc.js', criteria: { includes: [Array], excludes: [Array], basePath: '/home/nodebrick/application' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/home/nodebrick/application/node_modules/@typescript-eslint/parser/dist/parser.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js#overrides[0]', importerPath: '/home/nodebrick/application/.eslintrc.js' }, parserOptions: { ecmaVersion: 2018, project: [Array], tsconfigRootDir: './', sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'brace-style': 'off', camelcase: 'off', indent: 'off', 'no-empty-function': 'off', 'no-extra-parens': 'off', 'no-magic-numbers': 'off', 'no-unused-expressions': 'off', quotes: 'off', 'require-await': 'off', semi: 'off', 'space-before-function-paren': 'off', 'arrow-body-style': [Array], 'arrow-parens': [Array], 'capitalized-comments': 'off', 'no-multi-str': 'off', 'no-multiple-empty-lines': [Array], 'no-param-reassign': 'error', 'space-in-parens': [Array], '@typescript-eslint/adjacent-overload-signatures': 'error', '@typescript-eslint/array-type': [Array], '@typescript-eslint/await-thenable': 'error', '@typescript-eslint/ban-ts-ignore': 'off', '@typescript-eslint/ban-types': 'error', '@typescript-eslint/brace-style': [Array], '@typescript-eslint/camelcase': [Array], '@typescript-eslint/class-name-casing': 'error', '@typescript-eslint/consistent-type-assertions': 'error', '@typescript-eslint/consistent-type-definitions': [Array], '@typescript-eslint/explicit-function-return-type': 'error', '@typescript-eslint/explicit-member-accessibility': [Array], '@typescript-eslint/func-call-spacing': 'error', '@typescript-eslint/generic-type-naming': [Array], '@typescript-eslint/indent': [Array], '@typescript-eslint/interface-name-prefix': [Array], '@typescript-eslint/member-delimiter-style': [Array], '@typescript-eslint/member-naming': [Array], '@typescript-eslint/member-ordering': 'error', '@typescript-eslint/no-array-constructor': 'error', '@typescript-eslint/no-dynamic-delete': 'error', '@typescript-eslint/no-empty-function': 'error', '@typescript-eslint/no-empty-interface': 'error', '@typescript-eslint/no-explicit-any': 'error', '@typescript-eslint/no-extra-non-null-assertion': 'error', '@typescript-eslint/no-extra-parens': 'error', '@typescript-eslint/no-extraneous-class': [Array], '@typescript-eslint/no-floating-promises': 'error', '@typescript-eslint/no-for-in-array': 'error', '@typescript-eslint/no-inferrable-types': 'off', '@typescript-eslint/no-magic-numbers': 'off', '@typescript-eslint/no-misused-new': 'error', '@typescript-eslint/no-misused-promises': 'error', '@typescript-eslint/no-namespace': 'error', '@typescript-eslint/no-parameter-properties': 'error', '@typescript-eslint/no-require-imports': 'error', '@typescript-eslint/no-this-alias': 'error', '@typescript-eslint/no-type-alias': [Array], '@typescript-eslint/no-unnecessary-condition': 'error', '@typescript-eslint/no-unnecessary-qualifier': 'error', '@typescript-eslint/no-unnecessary-type-arguments': 'error', '@typescript-eslint/no-unnecessary-type-assertion': 'error', '@typescript-eslint/no-untyped-public-signature': 'error', '@typescript-eslint/no-unused-expressions': 'error', '@typescript-eslint/no-unused-vars-experimental': 'error', '@typescript-eslint/no-unused-vars': 'error', '@typescript-eslint/no-use-before-define': 'error', '@typescript-eslint/no-useless-constructor': 'error', '@typescript-eslint/no-var-requires': 'error', '@typescript-eslint/prefer-for-of': 'error', '@typescript-eslint/prefer-function-type': 'error', '@typescript-eslint/prefer-includes': 'error', '@typescript-eslint/prefer-namespace-keyword': 'error', '@typescript-eslint/prefer-nullish-coalescing': 'error', '@typescript-eslint/prefer-optional-chain': 'error', '@typescript-eslint/prefer-readonly': 'error', '@typescript-eslint/prefer-regexp-exec': 'error', '@typescript-eslint/prefer-string-starts-ends-with': 'error', '@typescript-eslint/promise-function-async': 'error', '@typescript-eslint/quotes': [Array], '@typescript-eslint/require-array-sort-compare': 'error', '@typescript-eslint/require-await': 'error', '@typescript-eslint/restrict-plus-operands': 'error', '@typescript-eslint/restrict-template-expressions': 'error', '@typescript-eslint/return-await': 'error', '@typescript-eslint/semi': [Array], '@typescript-eslint/space-before-function-paren': [Array], '@typescript-eslint/strict-boolean-expressions': 'off', '@typescript-eslint/triple-slash-reference': 'error', '@typescript-eslint/type-annotation-spacing': 'error', '@typescript-eslint/typedef': 'error', '@typescript-eslint/unbound-method': 'error', '@typescript-eslint/unified-signatures': 'error' }, settings: undefined }, { name: '--config » eslint:recommended', filePath: '/home/nodebrick/application/node_modules/eslint/conf/eslint-recommended.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'constructor-super': 'error', 'for-direction': 'error', 'getter-return': 'error', 'no-async-promise-executor': 'error', 'no-case-declarations': 'error', 'no-class-assign': 'error', 'no-compare-neg-zero': 'error', 'no-cond-assign': 'error', 'no-const-assign': 'error', 'no-constant-condition': 'error', 'no-control-regex': 'error', 'no-debugger': 'error', 'no-delete-var': 'error', 'no-dupe-args': 'error', 'no-dupe-class-members': 'error', 'no-dupe-keys': 'error', 'no-duplicate-case': 'error', 'no-empty': 'error', 'no-empty-character-class': 'error', 'no-empty-pattern': 'error', 'no-ex-assign': 'error', 'no-extra-boolean-cast': 'error', 'no-extra-semi': 'error', 'no-fallthrough': 'error', 'no-func-assign': 'error', 'no-global-assign': 'error', 'no-inner-declarations': 'error', 'no-invalid-regexp': 'error', 'no-irregular-whitespace': 'error', 'no-misleading-character-class': 'error', 'no-mixed-spaces-and-tabs': 'error', 'no-new-symbol': 'error', 'no-obj-calls': 'error', 'no-octal': 'error', 'no-prototype-builtins': 'error', 'no-redeclare': 'error', 'no-regex-spaces': 'error', 'no-self-assign': 'error', 'no-shadow-restricted-names': 'error', 'no-sparse-arrays': 'error', 'no-this-before-super': 'error', 'no-undef': 'error', 'no-unexpected-multiline': 'error', 'no-unreachable': 'error', 'no-unsafe-finally': 'error', 'no-unsafe-negation': 'error', 'no-unused-labels': 'error', 'no-unused-vars': 'error', 'no-useless-catch': 'error', 'no-useless-escape': 'error', 'no-with': 'error', 'require-atomic-updates': 'error', 'require-yield': 'error', 'use-isnan': 'error', 'valid-typeof': 'error' }, settings: undefined }, { name: '--config » plugin:@typescript-eslint/eslint-recommended', filePath: '/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '--config » plugin:@typescript-eslint/eslint-recommended#overrides[0]', filePath: '/home/nodebrick/application/node_modules/@typescript-eslint/eslint-plugin/dist/index.js', criteria: { includes: [Array], excludes: null, basePath: '/home/nodebrick/application' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'getter-return': 'off', 'no-dupe-args': 'off', 'no-dupe-keys': 'off', 'no-unreachable': 'off', 'valid-typeof': 'off', 'no-const-assign': 'off', 'no-new-symbol': 'off', 'no-this-before-super': 'off', 'no-undef': 'off', 'no-dupe-class-members': 'off', 'no-redeclare': 'off' }, settings: undefined }, { name: '--config', filePath: '/home/nodebrick/application/.eslintrc.js', criteria: null, env: { node: true, es6: true }, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2018 }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: {}, settings: undefined }, { name: '--config#overrides[0]', filePath: '/home/nodebrick/application/.eslintrc.js', criteria: { includes: [Array], excludes: [Array], basePath: '/home/nodebrick/application' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/home/nodebrick/application/node_modules/@typescript-eslint/parser/dist/parser.js', id: '@typescript-eslint/parser', importerName: '--config#overrides[0]', importerPath: '/home/nodebrick/application/.eslintrc.js' }, parserOptions: { ecmaVersion: 2018, project: [Array], tsconfigRootDir: './', sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'brace-style': 'off', camelcase: 'off', indent: 'off', 'no-empty-function': 'off', 'no-extra-parens': 'off', 'no-magic-numbers': 'off', 'no-unused-expressions': 'off', quotes: 'off', 'require-await': 'off', semi: 'off', 'space-before-function-paren': 'off', 'arrow-body-style': [Array], 'arrow-parens': [Array], 'capitalized-comments': 'off', 'no-multi-str': 'off', 'no-multiple-empty-lines': [Array], 'no-param-reassign': 'error', 'space-in-parens': [Array], '@typescript-eslint/adjacent-overload-signatures': 'error', '@typescript-eslint/array-type': [Array], '@typescript-eslint/await-thenable': 'error', '@typescript-eslint/ban-ts-ignore': 'off', '@typescript-eslint/ban-types': 'error', '@typescript-eslint/brace-style': [Array], '@typescript-eslint/camelcase': [Array], '@typescript-eslint/class-name-casing': 'error', '@typescript-eslint/consistent-type-assertions': 'error', '@typescript-eslint/consistent-type-definitions': [Array], '@typescript-eslint/explicit-function-return-type': 'error', '@typescript-eslint/explicit-member-accessibility': [Array], '@typescript-eslint/func-call-spacing': 'error', '@typescript-eslint/generic-type-naming': [Array], '@typescript-eslint/indent': [Array], '@typescript-eslint/interface-name-prefix': [Array], '@typescript-eslint/member-delimiter-style': [Array], '@typescript-eslint/member-naming': [Array], '@typescript-eslint/member-ordering': 'error', '@typescript-eslint/no-array-constructor': 'error', '@typescript-eslint/no-dynamic-delete': 'error', '@typescript-eslint/no-empty-function': 'error', '@typescript-eslint/no-empty-interface': 'error', '@typescript-eslint/no-explicit-any': 'error', '@typescript-eslint/no-extra-non-null-assertion': 'error', '@typescript-eslint/no-extra-parens': 'error', '@typescript-eslint/no-extraneous-class': [Array], '@typescript-eslint/no-floating-promises': 'error', '@typescript-eslint/no-for-in-array': 'error', '@typescript-eslint/no-inferrable-types': 'off', '@typescript-eslint/no-magic-numbers': 'off', '@typescript-eslint/no-misused-new': 'error', '@typescript-eslint/no-misused-promises': 'error', '@typescript-eslint/no-namespace': 'error', '@typescript-eslint/no-parameter-properties': 'error', '@typescript-eslint/no-require-imports': 'error', '@typescript-eslint/no-this-alias': 'error', '@typescript-eslint/no-type-alias': [Array], '@typescript-eslint/no-unnecessary-condition': 'error', '@typescript-eslint/no-unnecessary-qualifier': 'error', '@typescript-eslint/no-unnecessary-type-arguments': 'error', '@typescript-eslint/no-unnecessary-type-assertion': 'error', '@typescript-eslint/no-untyped-public-signature': 'error', '@typescript-eslint/no-unused-expressions': 'error', '@typescript-eslint/no-unused-vars-experimental': 'error', '@typescript-eslint/no-unused-vars': 'error', '@typescript-eslint/no-use-before-define': 'error', '@typescript-eslint/no-useless-constructor': 'error', '@typescript-eslint/no-var-requires': 'error', '@typescript-eslint/prefer-for-of': 'error', '@typescript-eslint/prefer-function-type': 'error', '@typescript-eslint/prefer-includes': 'error', '@typescript-eslint/prefer-namespace-keyword': 'error', '@typescript-eslint/prefer-nullish-coalescing': 'error', '@typescript-eslint/prefer-optional-chain': 'error', '@typescript-eslint/prefer-readonly': 'error', '@typescript-eslint/prefer-regexp-exec': 'error', '@typescript-eslint/prefer-string-starts-ends-with': 'error', '@typescript-eslint/promise-function-async': 'error', '@typescript-eslint/quotes': [Array], '@typescript-eslint/require-array-sort-compare': 'error', '@typescript-eslint/require-await': 'error', '@typescript-eslint/restrict-plus-operands': 'error', '@typescript-eslint/restrict-template-expressions': 'error', '@typescript-eslint/return-await': 'error', '@typescript-eslint/semi': [Array], '@typescript-eslint/space-before-function-paren': [Array], '@typescript-eslint/strict-boolean-expressions': 'off', '@typescript-eslint/triple-slash-reference': 'error', '@typescript-eslint/type-annotation-spacing': 'error', '@typescript-eslint/typedef': 'error', '@typescript-eslint/unbound-method': 'error', '@typescript-eslint/unified-signatures': 'error' }, settings: undefined } ] on /home/nodebrick/application/packages/nodebrick-database/src/models/enums +387ms
  eslint:ignored-paths contains: +427ms
  eslint:ignored-paths   target   = "/home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts" +1ms
  eslint:ignored-paths   base     = "/home/nodebrick/application" +0ms
  eslint:ignored-paths   relative = "packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts" +0ms
  eslint:ignored-paths   result   = false +0ms
  eslint:cli-engine Lint /home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts +0ms
  eslint:linter Linting code for /home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts +0ms
  eslint:linter Generating fixed text for /home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts (pass 1) +26s
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Complete iterating files: ["./packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts"] +26s
  eslint:cli-engine Linting complete in: 26083ms +26s

/home/nodebrick/application/packages/nodebrick-database/src/models/enums/CacheTypeEnum.ts
  2:1  error  Expected indentation of 4 spaces but found 0  @typescript-eslint/indent

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

~/application $ 
```

**Full .eslintrc.js**
```
module.exports = {
    //  as per https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        // "plugin:@typescript-eslint/recommended",
        // "plugin:@typescript-eslint/recommended-requiring-type-checking"
    ],
    "env": {
        "node": true,
        "es6": true,
    },
    parserOptions: {
        "ecmaVersion": 2018,
    },
    rules: {

    },

    overrides: [
        {
            //  use the typescript parser
            parser: "@typescript-eslint/parser",
            //  which tsconfig to use, the main and per packages ones
            parserOptions: {
                "ecmaVersion": 2018,
                "project": [
                    "./packages/*/tsconfig.eslint.json",
                ],
                "tsconfigRootDir": "./",
                "sourceType": "module"
            },
            plugins: [
                //  to be able to lint typescript files
                "@typescript-eslint"
            ],
            files: ["*.ts"],
            excludedFiles: ["*.js"],

            rules: {
                //  DEFINED IN TYPESCRIPT ESLINT
                "brace-style": "off", // defined in eslint ts - https://eslint.org/docs/rules/brace-style#require-brace-style-brace-style
                "camelcase": "off", // defined in eslint ts - https://eslint.org/docs/rules/camelcase#require-camelcase-camelcase
                "indent": "off", // defined in eslint ts - https://eslint.org/docs/rules/indent#enforce-consistent-indentation-indent
                "no-empty-function": "off",
                "no-extra-parens": "off",
                "no-magic-numbers": "off", // https://eslint.org/docs/rules/no-magic-numbers#disallow-magic-numbers-no-magic-numbers
                "no-unused-expressions": "off",
                "quotes": "off",
                "require-await": "off",
                "semi": "off",
                "space-before-function-paren": "off",

                //  ESLINT RULES
                "arrow-body-style": ["error", "as-needed"],
                "arrow-parens": ["error", "always"], // https://eslint.org/docs/rules/arrow-parens#require-parens-in-arrow-function-arguments-arrow-parens
                "capitalized-comments": "off", // https://eslint.org/docs/rules/capitalized-comments#enforce-or-disallow-capitalization-of-the-first-letter-of-a-comment-capitalized-comments
                "no-multi-str": "off", // https://eslint.org/docs/rules/no-multi-str
                "no-multiple-empty-lines": ["error", { "max": 1, "maxEOF": 0 }], // https://eslint.org/docs/rules/no-multiple-empty-lines#disallow-multiple-empty-lines-no-multiple-empty-lines
                "no-param-reassign": "error", // https://eslint.org/docs/rules/no-param-reassign
                "space-in-parens": ["error", "never"], // https://eslint.org/docs/rules/space-in-parens

                // TYPESCRIPT ESLINT RULES
                "@typescript-eslint/adjacent-overload-signatures": "error", //https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/adjacent-overload-signatures.md
                "@typescript-eslint/array-type": ["error", {"default": "array-simple"}], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/array-type.md
                "@typescript-eslint/await-thenable": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/await-thenable.md
                "@typescript-eslint/ban-ts-ignore": "off", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-ts-ignore.md
                "@typescript-eslint/ban-types": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md
                "@typescript-eslint/brace-style": ["error", "allman"], // already set in eslint - https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/brace-style.md
                "@typescript-eslint/camelcase": ["error"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/camelcase.md
                "@typescript-eslint/class-name-casing": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/class-name-casing.md
                "@typescript-eslint/consistent-type-assertions": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-assertions.md
                "@typescript-eslint/consistent-type-definitions": ["error", "interface"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-definitions.md
                "@typescript-eslint/explicit-function-return-type": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-function-return-type.md
                "@typescript-eslint/explicit-member-accessibility": [ // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/explicit-member-accessibility.md
                    "error",
                    {
                        "accessibility": "explicit",
                        "overrides": {
                            "accessors": "explicit",
                            "constructors": "explicit",
                            "parameterProperties": "explicit"
                        }
                    }
                ],
                "@typescript-eslint/func-call-spacing": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/func-call-spacing.md
                "@typescript-eslint/generic-type-naming": ["error", "^T[A-Z][a-zA-Z]+$"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/func-call-spacing.md
                "@typescript-eslint/indent": [ // https://eslint.org/docs/rules/indent#enforce-consistent-indentation-indent
                    "error",
                    4,
                    {
                        "ArrayExpression": "first",
                        "ObjectExpression": "first",
                        "FunctionDeclaration": {
                            "parameters": "first"
                        },
                        "FunctionExpression": {
                            "parameters": "first"
                        }
                    }
                ],
                "@typescript-eslint/interface-name-prefix": ["error", {"prefixWithI": "always"}], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/interface-name-prefix.md
                "@typescript-eslint/member-delimiter-style": [ // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-delimiter-style.md
                    "error",
                    {
                        "multiline": {
                            "delimiter": "semi",
                            "requireLast": true
                        },
                        "singleline": {
                            "delimiter": "semi",
                            "requireLast": false
                        }
                    }
                ],
                "@typescript-eslint/member-naming": ["error", {"private": "^_", "protected": "^_"}], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-naming.md
                "@typescript-eslint/member-ordering": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-ordering.md
                "@typescript-eslint/no-array-constructor": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-array-constructor.md
                "@typescript-eslint/no-dynamic-delete": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-dynamic-delete.md
                "@typescript-eslint/no-empty-function": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-function.md
                "@typescript-eslint/no-empty-interface": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-interface.md
                "@typescript-eslint/no-explicit-any": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-explicit-any.md
                "@typescript-eslint/no-extra-non-null-assertion": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-non-null-assertion.md
                "@typescript-eslint/no-extra-parens": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-parens.md
                "@typescript-eslint/no-extraneous-class": ["error", {"allowStaticOnly": true}], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extraneous-class.md
                "@typescript-eslint/no-floating-promises": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md
                "@typescript-eslint/no-for-in-array": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-for-in-array.md
                "@typescript-eslint/no-inferrable-types": "off", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-inferrable-types.md
                "@typescript-eslint/no-magic-numbers": "off", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-magic-numbers.md
                "@typescript-eslint/no-misused-new": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-misused-new.md
                "@typescript-eslint/no-misused-promises": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-misused-promises.md
                "@typescript-eslint/no-namespace": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-namespace.md
                "@typescript-eslint/no-parameter-properties": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-parameter-properties.md
                "@typescript-eslint/no-require-imports": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-require-imports.md
                "@typescript-eslint/no-this-alias": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-this-alias.md
                "@typescript-eslint/no-type-alias": ["error", {"allowAliases": "always"}], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-type-alias.md
                "@typescript-eslint/no-unnecessary-condition": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md
                "@typescript-eslint/no-unnecessary-qualifier": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-qualifier.md
                "@typescript-eslint/no-unnecessary-type-arguments": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-arguments.md
                "@typescript-eslint/no-unnecessary-type-assertion": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-assertion.md
                "@typescript-eslint/no-untyped-public-signature": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-untyped-public-signature.md
                "@typescript-eslint/no-unused-expressions": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-untyped-public-signature.md
                "@typescript-eslint/no-unused-vars-experimental": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md
                "@typescript-eslint/no-unused-vars": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
                "@typescript-eslint/no-use-before-define": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-use-before-define.md
                "@typescript-eslint/no-useless-constructor": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md
                "@typescript-eslint/no-var-requires": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-var-requires.md
                "@typescript-eslint/prefer-for-of": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-for-of.md
                "@typescript-eslint/prefer-function-type": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-function-type.md
                "@typescript-eslint/prefer-includes": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-includes.md
                "@typescript-eslint/prefer-namespace-keyword": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-namespace-keyword.md
                "@typescript-eslint/prefer-nullish-coalescing": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-nullish-coalescing.md
                "@typescript-eslint/prefer-optional-chain": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-optional-chain.md
                "@typescript-eslint/prefer-readonly": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-readonly.md
                "@typescript-eslint/prefer-regexp-exec": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-regexp-exec.md
                "@typescript-eslint/prefer-string-starts-ends-with": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-string-starts-ends-with.md
                "@typescript-eslint/promise-function-async": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/promise-function-async.md
                "@typescript-eslint/quotes": ["error", "double"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/quotes.md
                "@typescript-eslint/require-array-sort-compare": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/require-array-sort-compare.md
                "@typescript-eslint/require-await": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/require-await.md
                "@typescript-eslint/restrict-plus-operands": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-plus-operands.md
                "@typescript-eslint/restrict-template-expressions": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-template-expressions.md
                "@typescript-eslint/return-await": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md
                "@typescript-eslint/semi": ["error", "always"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md
                "@typescript-eslint/space-before-function-paren": ["error", "never"], // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/space-before-function-paren.md
                "@typescript-eslint/strict-boolean-expressions": "off", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md
                "@typescript-eslint/triple-slash-reference": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/triple-slash-reference.md
                "@typescript-eslint/type-annotation-spacing": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/type-annotation-spacing.md
                "@typescript-eslint/typedef": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/typedef.md
                "@typescript-eslint/unbound-method": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unbound-method.md
                "@typescript-eslint/unified-signatures": "error", // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unified-signatures.md
            }
        }
    ],
};
```

Thanks :)

--- a/packages/eslint-plugin/src/rules/brace-style.ts
+++ b/packages/eslint-plugin/src/rules/brace-style.ts
@@ -1,14 +1,16 @@
-import {
-  TSESTree,
-  AST_NODE_TYPES,
-} from '@typescript-eslint/experimental-utils';
+import { TSESTree } from '@typescript-eslint/experimental-utils';
 import baseRule from 'eslint/lib/rules/brace-style';
-import * as util from '../util';
+import {
+  InferOptionsTypeFromRule,
+  InferMessageIdsTypeFromRule,
+  createRule,
+  isTokenOnSameLine,
+} from '../util';
 
-export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
-export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+export type Options = InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
 
-export default util.createRule<Options, MessageIds>({
+export default createRule<Options, MessageIds>({
   name: 'brace-style',
   meta: {
     type: 'layout',
@@ -23,23 +25,117 @@ export default util.createRule<Options, MessageIds>({
   },
   defaultOptions: ['1tbs'],
   create(context) {
+    const [
+      style,
+      { allowSingleLine } = { allowSingleLine: false },
+    ] = context.options;
+
+    const isAllmanStyle = style === 'allman';
+    const sourceCode = context.getSourceCode();
     const rules = baseRule.create(context);
-    const checkBlockStatement = (
-      node: TSESTree.TSModuleBlock | TSESTree.TSInterfaceBody,
-    ): void => {
-      rules.BlockStatement({
-        type: AST_NODE_TYPES.BlockStatement,
-        parent: node.parent,
-        range: node.range,
-        body: node.body as any, // eslint-disable-line @typescript-eslint/no-explicit-any
-        loc: node.loc,
-      });
-    };
+
+    /**
+     * Checks a pair of curly brackets based on the user's config
+     */
+    function validateCurlyPair(
+      openingCurlyToken: TSESTree.Token,
+      closingCurlyToken: TSESTree.Token,
+    ): void {
+      if (
+        allowSingleLine &&
+        isTokenOnSameLine(openingCurlyToken, closingCurlyToken)
+      ) {
+        return;
+      }
+
+      const tokenBeforeOpeningCurly = sourceCode.getTokenBefore(
+        openingCurlyToken,
+      )!;
+      const tokenBeforeClosingCurly = sourceCode.getTokenBefore(
+        closingCurlyToken,
+      )!;
+      const tokenAfterOpeningCurly = sourceCode.getTokenAfter(
+        openingCurlyToken,
+      )!;
+
+      if (
+        !isAllmanStyle &&
+        !isTokenOnSameLine(tokenBeforeOpeningCurly, openingCurlyToken)
+      ) {
+        context.report({
+          node: openingCurlyToken,
+          messageId: 'nextLineOpen',
+          fix: fixer => {
+            const textRange: TSESTree.Range = [
+              tokenBeforeOpeningCurly.range[1],
+              openingCurlyToken.range[0],
+            ];
+            const textBetween = sourceCode.text.slice(
+              textRange[0],
+              textRange[1],
+            );
+
+            if (textBetween.trim()) {
+              return null;
+            }
+
+            return fixer.replaceTextRange(textRange, ' ');
+          },
+        });
+      }
+
+      if (
+        isAllmanStyle &&
+        isTokenOnSameLine(tokenBeforeOpeningCurly, openingCurlyToken)
+      ) {
+        context.report({
+          node: openingCurlyToken,
+          messageId: 'sameLineOpen',
+          fix: fixer => fixer.insertTextBefore(openingCurlyToken, '\n'),
+        });
+      }
+
+      if (
+        isTokenOnSameLine(openingCurlyToken, tokenAfterOpeningCurly) &&
+        tokenAfterOpeningCurly !== closingCurlyToken
+      ) {
+        context.report({
+          node: openingCurlyToken,
+          messageId: 'blockSameLine',
+          fix: fixer => fixer.insertTextAfter(openingCurlyToken, '\n'),
+        });
+      }
+
+      if (
+        isTokenOnSameLine(tokenBeforeClosingCurly, closingCurlyToken) &&
+        tokenBeforeClosingCurly !== openingCurlyToken
+      ) {
+        context.report({
+          node: closingCurlyToken,
+          messageId: 'singleLineClose',
+          fix: fixer => fixer.insertTextBefore(closingCurlyToken, '\n'),
+        });
+      }
+    }
 
     return {
       ...rules,
-      TSInterfaceBody: checkBlockStatement,
-      TSModuleBlock: checkBlockStatement,
+      'TSInterfaceBody, TSModuleBlock'(
+        node: TSESTree.TSModuleBlock | TSESTree.TSInterfaceBody,
+      ): void {
+        const openingCurly = sourceCode.getFirstToken(node)!;
+        const closingCurly = sourceCode.getLastToken(node)!;
+
+        validateCurlyPair(openingCurly, closingCurly);
+      },
+      TSEnumDeclaration(node): void {
+        const closingCurly = sourceCode.getLastToken(node)!;
+        const openingCurly = sourceCode.getTokenBefore(
+          node.members.length ? node.members[0] : closingCurly,
+        )!;
+
+        validateCurlyPair(openingCurly, closingCurly);
+      },
     };
   },
 });

--- a/packages/eslint-plugin/src/util/astUtils.ts
+++ b/packages/eslint-plugin/src/util/astUtils.ts
@@ -31,9 +31,20 @@ function isOptionalOptionalChain(
   );
 }
 
+/**
+ * Determines whether two adjacent tokens are on the same line
+ */
+function isTokenOnSameLine(
+  left: TSESTree.Token,
+  right: TSESTree.Token,
+): boolean {
+  return left.loc.end.line === right.loc.start.line;
+}
+
 export {
   LINEBREAK_MATCHER,
   isNotOptionalChainPunctuator,
   isOptionalChainPunctuator,
   isOptionalOptionalChain,
+  isTokenOnSameLine,
 };

--- a/packages/eslint-plugin/tests/rules/brace-style.test.ts
+++ b/packages/eslint-plugin/tests/rules/brace-style.test.ts
@@ -539,6 +539,38 @@ namespace Foo
       `,
       options: ['allman'],
     },
+    {
+      code: `
+enum Foo
+{
+  A,
+  B
+}
+      `,
+      options: ['allman'],
+    },
+    {
+      code: `
+enum Foo {
+  A,
+  B
+}
+      `,
+      options: ['1tbs'],
+    },
+    {
+      code: `
+enum Foo {
+  A,
+  B
+}
+      `,
+      options: ['stroustrup'],
+    },
+    {
+      code: `enum Foo { A, B }`,
+      options: ['1tbs', { allowSingleLine: true }],
+    },
   ],
 
   invalid: [
@@ -760,7 +792,6 @@ if (f) {
       options: ['allman'],
       errors: [{ messageId: 'sameLineClose' }],
     },
-
     // allowSingleLine: true
     {
       code: `function foo() { return; \n}`,
@@ -912,14 +943,12 @@ if (f) {
         { messageId: 'sameLineOpen' },
       ],
     },
-
     // Comment interferes with fix
     {
       code: `if (foo) // comment \n{\nbar();\n}`,
       output: null,
       errors: [{ messageId: 'nextLineOpen' }],
     },
-
     // https://github.com/eslint/eslint/issues/7493
     {
       code: `if (foo) {\n bar\n.baz }`,
@@ -995,7 +1024,6 @@ if (f) {
       options: ['allman'],
       errors: [{ messageId: 'sameLineOpen' }],
     },
-
     // https://github.com/eslint/eslint/issues/7621
     {
       code: `
@@ -1108,6 +1136,41 @@ namespace Foo {
       output: `namespace Foo \n{ \n }`,
       options: ['allman'],
       errors: [{ messageId: 'sameLineOpen' }],
+    },
+    {
+      code: `
+enum Foo
+{
+}
+      `,
+      output: `
+enum Foo {
+}
+      `,
+      errors: [{ messageId: 'nextLineOpen' }],
+    },
+    {
+      code: `
+enum Foo
+{
+}
+      `,
+      output: `
+enum Foo {
+}
+      `,
+      options: ['stroustrup'],
+      errors: [{ messageId: 'nextLineOpen' }],
+    },
+    {
+      code: `enum Foo { A }`,
+      output: `enum Foo \n{\n A \n}`,
+      options: ['allman'],
+      errors: [
+        { messageId: 'sameLineOpen' },
+        { messageId: 'blockSameLine' },
+        { messageId: 'singleLineClose' },
+      ],
     },
   ],
 });

